### PR TITLE
Add demo for shipping Vector internal logs/metrics to Datadog

### PR DESCRIPTION
**Pull request by lucperkins**

### CODE REVIEW

File: datadog-observability/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+In this demo, Vector is running a simple topology in which the `generator` source emits random JSON
+logs twice per second and sends those logs off to stdout via the `console` sink. But even this
+simple pipeline is enough to generate internal logs and metrics for Vector (via the eponymous
+sources).
+
+To run this scenario, make sure you have the `DD_API_KEY` environment variable set to your Datadog
+API key and then run:
+
+```bash
+docker compose up --detach --remove-orphans
```
Comment (jdrouet) [inline datadog-observability/README.md:12]: we could remove the `--remove-orphans` to kill this simple (and avoid removing any non-related container)

**Review by jdrouet — COMMENTED** [review]
I'm wondering if we could merge the datadog examples into a single one to avoid people to be confused.
Or remove the `datadog` sink from the `datadog-agent` example 🤔

**Review by spencergilbert — APPROVED** [review]
I think we should determine some more guidelines (generally) of what should be added to the demos repo, if only to control sprawl and keep demos useful

**Comment by lucperkins** [comment]
I'm going to merge this as-is for now, as I can't get the `datadog-agent` demo to work with the Datadog API (not sure why, but noted in #8). We can merge the scenarios later if that seems like the right thing to do.

**Pull request merged**